torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -4009,6 +4011,7 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::barrier(const BarrierOptions& opts) {
   RECORD_PARAM_COMMS(
       static_cast<int>(
           this->getSequenceNumberForGroup() + 1), // seq + 1 to match collective
```

Comment:
seems we should remove the function `getSequenceNumberForGroup()`, but use `seqCollective_` directly.
1. func name can't represent it returns collective sequence id;
2. keep conssistant with P2P, which use `seqP2P` directly;

Or we can reimplement `getSequenceNumberForGroup()` to:
```
uint64_t ProcessGroupNCCL::getSequenceNumberForGroup(OpType typ, bool batchP2P = false) {
  return isP2P(typ, batchP2P) ? seqP2P_ : seqCollective_;
}
```

Reply:
getSequenceNumberForGroup is a virtual function defined in root class BackEnd. P2P seems only used in ProcessGroupNCCL. Add a new input parameter probably does not make sense to other sub-classes.

Reply:
got it. Then how about using `seqCollective_` directly?